removed warnings of clang@11 compiler

ravil-mobile opened the pull request:
* added submodules/async via system system
* added -Wno-unused-parameter flag for clang

ravil-mobile commented:
@krenzland , can you have a look at ?
```
ODEVector.h:86:14: warning: definition of implicit copy constructor for 'ODEVector' is deprecated because it has a user-provided copy assignment operator
```

sebwolf-de commented on src/Initializer/MemoryAllocator.cpp:
```diff
@@ -79,7 +79,7 @@
 void* seissol::memory::allocate(size_t i_size, size_t i_alignment, enum Memkind i_memkind)
 {
-    void* l_ptrBuffer;
+    void* l_ptrBuffer{};
```
Maybe `=nullptr` would be more verbose?

krenzland replied:
Imo that is clear enough, nulptr is obviously the default value

ravil-mobile commented:
Hi, I decided to remove all `NULL` from `seissol::memory::allocate` replace them with `nullptr`.

krenzland commented:
> @krenzland , can you have a look at ?
> 
> ```
> ODEVector.h:86:14: warning: definition of implicit copy constructor for 'ODEVector' is deprecated because it has a user-provided copy assignment operator
> ```

Hm, it probably means that we need to define a copy constructor manually for this class?

ravil-mobile commented:
Removed `ODEVector.h:86:14: warning: definition of implicit copy constructor for 'ODEVector' is deprecated because it has a user-provided copy assignment operator`

@krenzland, I just added
```
ODEVector(const ODEVector&) = default;
```
because all methods attributes have their defined copy-constructors
